### ISSUE TITLE
LS diagnostic not sent for standalone Java file

### ISSUE THREAD

**Issue by mickaelistria** [redhat-developer/eclipseide-jdtls#103]
Let's make sure issue is not already fixed in snapshots first.

- [x] I verified I can reproduce this issue against latest [snapshots build](https://github.com/redhat-developer/eclipseide-jdtls#-installation)

## Steps to reproduce

From a fresh installation and clean workspace:

* File > Open File > browse to a basic functioning Java file
* Verify hover, completion... to verify LS is functioning

I tried

* insert a random wrong character at some point

I expected: an error report about the syntax error

But got: nothing

## Here is some relevant log output form (`<workspace>/.metadata/.log`)

The language server logs show that JDT-LS does *not* send any diagnostics for that case.

## Tested under this environment:
* OS & version: Fedora Latest
* Eclipse IDE/Platform version: Latest I-Build
* ...

## Community

- [x] I understand reporting an issue to this OSS project does not mandate anyone to fix it. Other contributors may consider the issue, or not, at their own convenience. The most efficient way to get it fixed is that I fix it myself and contribute it back as a good quality patch to the project.

**Comment by mickaelistria** [comment]
Progress note: a patch in JDT-LS can help, but I'm not sure it's the best fix yet.

In BaseDocumentLifecycleHandler, replace
```java
		List<ICompilationUnit> validateCopy =
			preferenceManager.getPreferences().isValidateAllOpenBuffersOnChanges()
				? Arrays.asList(JavaCore.getWorkingCopies(null)) : new ArrayList<>(toValidate);

```
with
```java
		LinkedHashSet<ICompilationUnit> validateCopy = new LinkedHashSet<>(toValidate);
		if (preferenceManager.getPreferences().isValidateAllOpenBuffersOnChanges()) {
			toValidate.addAll(Arrays.asList(JavaCore.getWorkingCopies(null)));
		}
```
to force dirty documents to be validated in any case.

**Comment by mickaelistria** [comment]
I think the fix for #104 does cover this problem. But still I submitted https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2886 to just let diagnostics flow on open (instead of having to go through an edit)

**Comment by mickaelistria** [comment]
This seems fixed with recent changes in JDT-LS.
